tests/fixture/test_data_server.go

```diff
@@ -37,15 +38,14 @@ type TestData struct {
 
 // http server allocating resources to tests potentially executing in parallel
 type testDataServer struct {
-	TestData
-
 	// Synchronizes access to test data
 	lock sync.Mutex
+	TestData
 }
 
 // Type used to marshal/unmarshal a set of test keys for transmission over http.
 type keysDocument struct {
-	Keys []*secp256k1.PrivateKey
+	Keys []*secp256k1.PrivateKey `json:"keys"`
```

Comment:
I know the marshaller defaults to the name of the variable... But I think explicitly tagging makes it very clear that we are json marshalling this. Also we typically use `camelCase` for json rather than `CamelCase`.